capa/features/extractors/vmray/__init__.py

```diff
@@ -0,0 +1,165 @@
+import logging
+from typing import Dict, List, Tuple, Optional
+from pathlib import Path
+from zipfile import ZipFile
+from collections import defaultdict
+
+from capa.exceptions import UnsupportedFormatError
+from capa.features.extractors.vmray.models import File, Flog, SummaryV2, StaticData, FunctionCall, xml_to_dict
+
+logger = logging.getLogger(__name__)
+
+# TODO (meh): is default password "infected" good enough?? https://github.com/mandiant/capa/issues/2148
+DEFAULT_ARCHIVE_PASSWORD = b"infected"
+
+SUPPORTED_FLOG_VERSIONS = ("2",)
+
+
+class VMRayAnalysis:
+    def __init__(self, zipfile_path: Path):
+        self.zipfile = ZipFile(zipfile_path, "r")
+
+        # summary_v2.json is the entry point to the entire VMRay archive and
+        # we use its data to find everything else that we need for capa
+        self.sv2 = SummaryV2.model_validate_json(
+            self.zipfile.read("logs/summary_v2.json", pwd=DEFAULT_ARCHIVE_PASSWORD)
+        )
+        self.file_type: str = self.sv2.analysis_metadata.sample_type
+
+        # flog.xml contains all of the call information that VMRay captured during execution
+        flog_xml = self.zipfile.read("logs/flog.xml", pwd=DEFAULT_ARCHIVE_PASSWORD)
+        flog_dict = xml_to_dict(flog_xml)
+        self.flog = Flog.model_validate(flog_dict)
+
+        if self.flog.analysis.log_version not in SUPPORTED_FLOG_VERSIONS:
+            logger.warning("VMRay feature extractor does not support flog version %s", self.flog.analysis.log_version)
+            raise UnsupportedFormatError(
+                "VMRay feature extractor does not support flog version %s", self.flog.analysis.log_version
+            )
+
+        self.exports: Dict[int, str] = {}
+        self.imports: Dict[int, Tuple[str, str]] = {}
+        self.sections: Dict[int, str] = {}
+        self.process_ids: Dict[int, int] = {}
+        self.process_threads: Dict[int, List[int]] = defaultdict(list)
+        self.process_calls: Dict[int, Dict[int, List[FunctionCall]]] = defaultdict(lambda: defaultdict(list))
+        self.base_address: int
+
+        self.sample_file_name: Optional[str] = None
+        self.sample_file_analysis: Optional[File] = None
+        self.sample_file_static_data: Optional[StaticData] = None
+
+        self._find_sample_file()
+
+        if self.sample_file_name is None or self.sample_file_analysis is None:
+            logger.warning("VMRay archive does not contain sample file (file_type: %s)", self.file_type)
+            raise UnsupportedFormatError("VMRay archive does not contain sample file (file_type: %s)", self.file_type)
+
+        if not self.sample_file_static_data:
+            # we see this for text files e.g. JScript file types
+            logger.warning("VMRay archive does not contain static data (file_type: %s)", self.file_type)
+            raise UnsupportedFormatError("VMRay archive does not contain static data (file_type: %s)", self.file_type)
```

Comment:
should we just explicitly check `self.file_type` instead/in addition to all these other checks?

Reply:
I've experienced VMRay analysis archives in various shapes and sizes. The file type does not definitively tell us what data we can expect to find in the archive, so to be explicit we check for the various pieces that we need at a minimum to run capa analysis.

Reply:
ah great, let's add this as a comment in the source